lib/Frontend/Frontend.cpp

```diff
@@ -1065,6 +1079,13 @@ void CompilerInstance::performSema() {
 }
 
 bool CompilerInstance::loadStdlibIfNeeded() {
+  if (!FrontendOptions::doesActionRequireSwiftStandardLibrary(
+          Invocation.getFrontendOptions().RequestedAction)) {
+    return false;
+  }
+  if (Invocation.getFrontendOptions().InputsAndOutputs.shouldTreatAsLLVM()) {
+    return false;
+  }
```

Comment:
Can these checks be sunk down into `CompilerInvocation::getImplicitStdlibKind`?

Reply:
I think it would make sense, but I’m currently hitting issues when scanning dependencies because apparently dependency scanning requires the stdlib to be loaded but returns fails in `FrontendOptions::doesActionRequireSwiftStandardLibrary` (see other comment).

Reply:
I actually realize now `doesActionRequireSwiftStandardLibrary` shouldn't be sunk into `getImplicitStdlibKind`, see my other comment